tests/conftest.py

```diff
@@ -111,3 +118,37 @@ def skip_by_profile_type(request):
         for skip_profile_type in request.node.get_closest_marker("skip_profile").args:
             if skip_profile_type == profile_type:
                 pytest.skip(f"skipped on '{profile_type}' profile")
+
+
+# solution from https://github.com/pytest-dev/pytest-xdist/issues/385#issuecomment-1176976154
+class XDistSerialPlugin:
+    def __init__(self):
+        self._nodes = None
+
+    @pytest.hookimpl(tryfirst=True)
+    def pytest_collection(self, session):
+        if is_xdist_controller(session):
+            self._nodes = {
+                item.nodeid: item
+                for item in session.perform_collect(None)
+            }
+            return True
+
+    def pytest_xdist_make_scheduler(self, config, log):
+        return SerialScheduling(config, log, nodes=self._nodes)
+
+
+class SerialScheduling(LoadScopeScheduling):
```

Comment:
The comments in this class make it sound as if this will allow, or is accounting for, the test classes in `test_python_model.py` to reference each other, or artifacts created by each other. Am I understanding that correctly?

Reply:
It was intended to put all of them in the same runner when we have multiple of them so that all python model tests run in sequence and we don't over load the cluster.

Reply:
Ah ok, so "scope" here is not referring to variable scope, but instead to the resources to which these tests have access?

Reply:
I think so, I mostly copy pasted the code from stackoverflow without fully understood what it was doing.